php-frontend/src/main/java/org/sonar/php/tree/impl/SeparatedListImpl.java

```diff
@@ -0,0 +1,210 @@
+    Preconditions.checkArgument(
+      list.size() == separators.size() + 1 || list.size() == separators.size(),
+      "Instanciating a SeparatedList with inconsistent number of elements (%s) and separators (%s)",
+      list.size(), separators.size());
+    this.list = list;
+    this.separators = separators;
+  }
+
+  public static <T extends Tree> SeparatedListImpl<T> empty() {
+    return new SeparatedListImpl<>(ImmutableList.<T>of(), ImmutableList.<InternalSyntaxToken>of());
+  }
+
+  public InternalSyntaxToken getSeparator(int i) {
+    return separators.get(i);
+  }
+
+  public List<InternalSyntaxToken> getSeparators() {
+    return separators;
+  }
+
+
+  @Override
+  public int size() {
+    return list.size();
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return list.isEmpty();
+  }
+
+  @Override
+  public boolean contains(Object o) {
+    return list.contains(o);
+  }
+
+  @Override
+  public Iterator<T> iterator() {
+    return list.iterator();
+  }
+
+  @Override
+  public Object[] toArray() {
+    return list.toArray();
+  }
+
+  @Override
+  public <T> T[] toArray(T[] a) {
+    return list.toArray(a);
+  }
+
+  @Override
+  public boolean add(T e) {
+    return list.add(e);
+  }
+
+  @Override
+  public boolean remove(Object o) {
+    return list.remove(o);
+  }
+
+  @Override
+  public boolean containsAll(Collection<?> c) {
+    return list.containsAll(c);
+  }
+
+  @Override
+  public boolean addAll(Collection<? extends T> c) {
+    return list.addAll(c);
+  }
+
+  @Override
+  public boolean addAll(int index, Collection<? extends T> c) {
+    return list.addAll(index, c);
+  }
+
+  @Override
+  public boolean removeAll(Collection<?> c) {
+    return list.removeAll(c);
+  }
+
+  @Override
+  public boolean retainAll(Collection<?> c) {
+    return list.retainAll(c);
+  }
+
+  @Override
+  public void clear() {
+    list.clear();
+  }
+
+  @Override
+  public T get(int index) {
+    return list.get(index);
+  }
+
+  @Override
+  public T set(int index, T element) {
+    return list.set(index, element);
+  }
+
+  @Override
+  public void add(int index, T element) {
+    list.add(index, element);
+  }
+
+  @Override
+  public T remove(int index) {
+    return list.remove(index);
+  }
+
+  @Override
+  public int indexOf(Object o) {
+    return list.indexOf(o);
+  }
+
+  @Override
+  public int lastIndexOf(Object o) {
+    return list.lastIndexOf(o);
+  }
+
+  @Override
+  public ListIterator<T> listIterator() {
+    return list.listIterator();
+  }
+
+  @Override
+  public ListIterator<T> listIterator(int index) {
+    return list.listIterator(index);
+  }
+
+  @Override
+  public List<T> subList(int fromIndex, int toIndex) {
+    return list.subList(fromIndex, toIndex);
+  }
+
+  public Iterator<Tree> elementsAndSeparators(final Function<T, ? extends Tree> elementTransformer) {
+    return new ElementAndSeparatorIterator(elementTransformer);
+  }
+
+  public Iterator<Tree> elementsAndSeparators() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)